cli/flox-test-utils/src/proptest.rs

```diff
@@ -39,11 +69,126 @@ pub fn empty_btree_map_alphanum_keys(
     key_max_size: usize,
     max_keys: usize,
 ) -> impl Strategy<Value = BTreeMap<String, BTreeMap<(), ()>>> {
-    use prop::collection::btree_map;
-
-    prop::collection::btree_map(
+    btree_map(
         alphanum_string(key_max_size),
         btree_map(any::<()>(), any::<()>(), 0),
         0..max_keys,
     )
 }
+
+/// Produces `Vec<String>` instances with limited sizes for performance reasons.
+pub fn vec_of_strings(
+    string_max_size: usize,
+    max_elements: usize,
+) -> impl Strategy<Value = Vec<String>> {
+    prop_vec(alphanum_string(string_max_size), 0..=max_elements)
```

Comment:
nonblocking question: would it be better to expand the character set and decrease the length of the string? Seems like that might be more likely to surface an edge case

Reply:
That depends on what you're using the generated output for. If you just need a string to be a key in a map, then no, your suggestion wouldn't really catch any more edge cases. If you're using this output as the name of a package or environment, then sure, but we already have a different strategy for that: https://github.com/flox/flox/blob/4ae9ef0ce14a745ead83c98f3a072cd1ec4b261c/cli/flox-rust-sdk/src/models/environment_ref.rs#L152